### ISSUE TITLE
Toggle groups are finicky, and should probably be a different feature entirely.

### ISSUE THREAD

**Issue by FremyCompany** [tabatkins/css-toggle#10]
In the following example:

<blockquote>
<details><summary><b>Example from the spec</b>

in which a `panel-set` contains multiple `panel-tab` only one of which can be active at the same time, thanks to the `group` feature

</summary>

```html
<panel-set>
  <panel-tab>first tab</panel-tab>
  <panel-card>first panel</panel-card>
  <panel-tab>second tab</panel-tab>
  <panel-card>second card</panel-card>
  ...
</panel-set>
<style>
panel-set {
  /* The common ancestor sets up a group */
  toggle-group: tab;
}
panel-tab {
  /* Each tab creates a sticky toggle
    (so once it’s open, clicking again won’t close it),
    opts into the group,
    and declares itself a toggle activator */
  toggle: tab 1 group sticky;
}
panel-tab:first-of-type {
  /* The first tab also sets its initial state
    to be active */
  toggle: tab 1/1 group sticky;
}
panel-tab:toggle(tab) {
  /* styling for the active tab */
}
panel-card {
  /* cards are hidden by default */
  display: none;
}
panel-card:toggle(tab) {
  display: block;
}
</style>
```

</details>
</blockquote>

What happens if the element that is the active toggle gets removed by script? Will this lead to a situation where no panel-card will be opened anymore? This doesn't feel like the right behavior for a tabbed interface.

Similarly, what happens if a new tab and card are inserted as first elements in the panel? The toggle for that new element is supposed to be set as 'active' per the css rule, unactivating the previous tab (I would think).

Based on this and the other issue I plan to file about what I perceive as accessibility issues with the current design, I'm wondering whether it wouldn't make more sense to take an approach where grouped-toggles are no longer allowed, and rather are moved to a different feature, called `radio-group`. 

A `radio-group` has a name and a selection algorithm (when it's set to `default-to-first`, the group must always have a `:checked` radio element, and if no currently eligible element is active, the first such inactive element gets activated on the spot; `default-to-last`: works similarly but selects the last element in dom order instead; `no-default` enables a state where no element is selected in the group)

Then the code can be:
```css
panel-set {
    radio-group: tab default-to-first;
}
panel-tab {
    radio-group-element: tab;
}
panel-card {
    display: none;
}
panel-tab:checked {
    /* active state */
}
panel-tab:checked + panel-card {
    display: block;
}
```

This is, in my opinion, both easier to understand and more semantic.

**Comment by FremyCompany** [comment]
I supposed that to avoid overloading it, we could decide to rename the `:checked` pseudo-class above into `:radio-active(tab)` ☢️ or `:active-radio(tab)` 📻 or something similar

**Comment by tabatkins** [comment]
> What happens if the element that is the active toggle gets removed by script? Will this lead to a situation where no panel-card will be opened anymore? This doesn't feel like the right behavior for a tabbed interface.

Yes, if you remove the element that is currently active, there won't be any active toggles. This is how radio groups work today, for example. It won't stop you from activating one of the remaining toggles, tho.

> Similarly, what happens if a new tab and card are inserted as first elements in the panel? The toggle for that new element is supposed to be set as 'active' per the css rule, unactivating the previous tab (I would think).

It's currently unspecified what happens if a new toggle is created in the group with an active value, and there are several possibilities:

1) Treat it like it was triggered to the starting value (thus unsetting the existing active toggle)
2) Simply initialize it as specified without triggering, so there might be multiple active elements until a trigger does come thru and rationalizes the values.
3) When a new toggle would be part of a group, immediately apply the "only one active" rule as if the element was around for the most recent trigger (thus setting the new toggle to 0).

I suspect (3) is the best way to do this; it feels like the most expected behavior.

> I'm wondering whether it wouldn't make more sense to take an approach where grouped-toggles are no longer allowed, and rather are moved to a different feature, called radio-group.

As far as I can tell, this is *exactly the same behavior* as the toggle-* properties, just renamed and with the currently-unspecified behavior specified in some way. (And note that you still need the `toggle-trigger` equivalent, too; that's not mentioned in your example.)

The one new feature seems to be the ability to *force* one toggle to be active, rather than just enforcing that *at most* one toggle is active. Notably, *this is not how radio buttons works*, so calling it "radio-*" probably isn't great. ^_^  We could just add this functionality to `toggle-group`, tho.

Is there any other reason why you think the toggle-group functionality should be moved to a new name, when it still has to exactly duplicate the existing complete set of toggle-* functionality?

**Comment by FremyCompany** [comment]
You don't need to specify `toggle-trigger` because an element that is a `radio-group-element` has a trigger automatically to become the active one once activated, in the current design. The mechanism behind this would be similar to toggle, but cleanly separated so that the semantic defaults make sense for the use case.

That said, my main reason for wanting a separate system is that, from an accessibility point of view, I'm unhappy with the idea to use "heuristics" to define behaviors that are significantly different from one another (toggles vs radio). It is very easy for authors to "update" the system to deal with an exception case ("well, maybe this one should have the value 2 if it's the selected element by default and not by click") and break the accessiblity entirely. I will elaborate this at some point, but I don't have the energy to look up and reference all the ARIA specs, but I eventually will.

For this reason, I would rather have toggles and radios be two different things.

The other reason I would want that, is that it's super confusing to have to set values like `1/1 group sticky`. It's not clear what that does, while the behavior that is desired is rather straightfoward. Specifying the fallback algorithm (`default-to-first`/...) on the element that defines the group is also very logical as a choice, and avoids to write a special rule to select the first default element, that is creating finicky issues later on (yes, we can of course resolve them in some ways, but this is not going to be obvious to authors either).

**Comment by FremyCompany** [comment]
In more details, I would rather have three interfaces for toggles.

1. `radio-group` (or select-group or whatever) where one only needs to define 
     - the group name and element scope
     - which elements are `radio-group-element`s
     - which elements are affected by the tab visibility state
2. `state` which is a value that is defined on the root element only, and can be updated by trigger to either 
     - increment the value
     - decrement the value
     - set the value
     descendants of elements defining a state can have trigger for that state, but it is stored on the root, not on that element.

**Comment by FremyCompany** [comment]
But I guess my main complain is that a `true`/`false` that define a unique (or no) active state is fudamentally different from a system where you can define a value that is non-binary. These things should be exposed with different accessibility values.

In one case, you have a `state` that is a number; in the other case, you have one element that is the active descendant (or none). It is not just a number, and in fact cannot be.
